File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        p = [0] * len(nums)
        s = [0] * len(nums)
        p[0] = nums[0]
        s[-1] = nums[-1]
        out = [0] * len(nums)
        for i in range(1,len(nums)):
            p[i] = p[i-1] * nums[i]
            s[-i-1] = s[-i] * nums[-i-1]
        out[0] = s[1]
        out[-1] = p[-2]
        for i in range(1,len(nums)-1):
            out[i] = p[i-1] * s[i+1]
        return out

File: Data Structures & Algorithms/three-integer-sum/submission-0.cpp
class Solution {
public:
    vector<vector<int>> threeSum(vector<int>& nums) {
        int temp;
        for(int i=0;i<nums.size();i++){
            for(int j=0;j<nums.size()-i-1;j++){
                if(nums[j]>nums[j+1]){
                    temp = nums[j];
                    nums[j] = nums[j+1];
                    nums[j+1] = temp;
                }
            }
        }
        set<vector<int>> r1;
        for(int i=0;i<nums.size()-2;i++){
            for(int j=i+1;j<nums.size()-1;j++){
                for(int k=j+1;k<nums.size();k++){
                    if(0==nums[i]+nums[j]+nums[k])
                    r1.insert({nums[i],nums[j],nums[k]});
                }
            }
        }
        vector<vector<int>> result(r1.begin(),r1.end());
        return result;
    }
};
